File: C/day053.c
/* Author: Ahmed A. M. ALOMAR
Date: February 22 - 2020
Description:
This program is an example of using the function call */


#include <stdio.h>

int adding_numbers(int x, int y);


int main(void)
{
  int result = 0;
  result = adding_numbers(20, 40);

  printf("Function return results: %d \n", result);

  printf("Printing a function call with 20, 20 parameters: %d \n", adding_numbers(20, 20));

  int box = adding_numbers(50, 50);
  printf("Using the function to produce a result as a value for a new variable box: %d \n", box);

  return 0;
}


int adding_numbers(int x, int y)
{
  int result = 0;
  result = x + y;

  return result;
}

File: SourceCode/Python/day053.py
# ------------------------------------ #
# Author: Ahmed A. M. ALOMAR
# Date: June 2 - 2020
#
# Description:
# * Example of Return function as an 
#   argument
# ------------------------------------ #



def calculator(operation):
  def add(x, y):
    return x + y
  def subtract(x, y):
    return x - y

  if operation == "add":
    return add
  elif operation == "subtract":
    return subtract


print(calculator("add")(10, 4))
print(calculator("subtract")(7, 7))

